service/labelstore/service.go

```diff
@@ -196,26 +198,23 @@ func (s *service) GetLocalRefID(componentID string, globalRefID uint64) uint64 {
 	return local
 }
 
-// AddStaleMarker adds a stale marker
-func (s *service) AddStaleMarker(globalRefID uint64, l labels.Labels) {
+func (s *service) TrackStaleness(ids []StalenessTracker) {
 	s.mut.Lock()
 	defer s.mut.Unlock()
```

Comment:
Just a thought: could we use a different mutex for `s.staleGlobals` and thus avoid contention with other methods like GetLocalRefID, GetGlobalRefID, GetOrAddGlobalRefID, etc? seems possible at a glance

Reply:
I fiddled with that a bit but it felt a bit more error prone in developing. I would prefer to split that out to another pr if we wanted to go that route.